rpc/rpcserver/server.go

```diff
@@ -481,18 +481,19 @@ func (s *walletServer) NextAddress(ctx context.Context, req *pb.NextAddressReque
 		return nil, translateError(err)
 	}
 
-	pubKey, err := s.wallet.PubKeyForAddress(addr)
-	if err != nil {
-		return nil, translateError(err)
-	}
-	pubKeyAddr, err := dcrutil.NewAddressSecpPubKey(pubKey.Serialize(), s.wallet.ChainParams())
-	if err != nil {
-		return nil, translateError(err)
+	var pubKeyAddrString string
+	if secp, ok := addr.(wallet.SecpPubKeyer); ok {
```

Comment:
Isn't this changing the error handling semantics?  Previously, if it got to this point and the address can't be parsed into a pubkey, it would return an error.  Now, it will ignore it and return what is almost surely an invalid address.

Reply:
Correct, but this method should still work on accounts even if the addresses represented by the account are not p2pkh.